room/compiler-processing/src/main/java/androidx/room/compiler/processing/ksp/KspRoundEnv.kt

```diff
@@ -37,4 +42,37 @@ internal class KspRoundEnv(
                 env.wrapClassDeclaration(it)
             }.toSet()
     }
+
+    override fun getTypeElementsAnnotatedWith(klass: KClass<out Annotation>): Set<XTypeElement> {
+        return getTypeElementsAnnotatedWith(
+            annotationQualifiedName = klass.qualifiedName ?: error("No qualified name for $klass")
+        )
+    }
+
+    override fun getElementsAnnotatedWith(klass: KClass<out Annotation>): Set<XElement> {
+        return getElementsAnnotatedWith(
+            annotationQualifiedName = klass.qualifiedName ?: error("No qualified name for $klass")
+        )
+    }
+
+    override fun getElementsAnnotatedWith(annotationQualifiedName: String): Set<XElement> {
```

Comment:
this is not related to xprocessing but it might make sense to add some tests w/ AnnotationTargets to ensure KSP handles these gracefuly (and if not file a bug).

Reply:
What is the ideal way for KSP to handle a wrong target? I tried adding a unit test, and compilation does fail because of that, but it seems like KSP does run first before compilation failure (at least in the xprocessing test harness), which could cause confusing errors.